### DISCUSSION TITLE
Undefined Reference Errors When Using Demo

### DISCUSSION THREAD

**Discussion by NyquilDreams** [trilinos/Trilinos#13195] [Q&A] [answered]
When making sample code with the files provided in the simpleBuildAgainstTrilinos folder, I ran into undefined reference errors on compilation. Manually inserting `/ascldap/users/ihalim/internship/Trilinos-install/lib64/libmuelu.so.15.2.0
/ascldap/users/ihalim/internship/Trilinos-install/lib64/libxpetra.so.15.2.0
/ascldap/users/ihalim/internship/Trilinos-install/lib64/libxpetra-sup.so.15.2.0` into the link.txt file seems to have solved the problem, but I'm left wondering why they weren't being linked in the first place. Any ideas?

@jhux2

**Comment by jhux2** [comment]
@bartlettroscoe @sebrowne Any ideas/suggestions for Ian?

**Comment by bartlettroscoe** [comment]
@NyquilDreams, there are automated Trilinos tests that install Trilinos and then build the simpleBuildAgainstTrilinos CMake project against it.  You can see these tests at:

https://github.com/trilinos/Trilinos/blob/d2ba0c30429c2a72032f3be3f151eeff126e553a/packages/TrilinosInstallTests/CMakeLists.txt#L226-L261

I wish I could show you the results of that test running on CDash but Trilinos does not currently have public CDash site to submit results to (so you will have to take my word for it that this test passes in all of the automated Trilinos builds).

Can you please attach your Trilinos configure script calling cmake, the STDOUT+STDERR output from the `cmake ...`, and the generated CMakeLists.txt file?  I may be able to see what the issue is from there.

**Comment by NyquilDreams** [comment, marked as answer]
It turns out that CMakeLists.txt was simply missing MueLu and Xpetra in the list of components. Editing this line fixed the issue. `find_package(Trilinos REQUIRED COMPONENTS Teuchos Tpetra MueLu Xpetra)`

**Reply by bartlettroscoe** [reply]
simpleBuildAgainstTrilinos  does not depend on Xpetra or MueLu, does it?

**Reply by NyquilDreams** [reply]
> simpleBuildAgainstTrilinos does not depend on Xpetra or MueLu, does it?

It doesn't, but I had changed app.cpp in a way that did require them.
